node/src/components/consensus/highway_core/state.rs

```diff
@@ -503,6 +498,35 @@ impl<C: Context> State<C> {
         }
         equivocators
     }
+
+    /// Returns an error if the vote's round exponent was changed away from the cited votes' median
+    /// by more than the allowed spread.
+    fn check_round_exp_spread(&self, wvote: &WireVote<C>) -> Result<(), VoteError> {
+        let panorama = &wvote.panorama;
+        if let Observation::Correct(vh) = &panorama[wvote.creator] {
+            if wvote.round_exp >= self.vote(vh).round_exp {
+                return Ok(()); // Round exponent was not decreased.
+            }
+        }
+        let upper_re = match wvote.round_exp.checked_add(self.params.round_exp_spread()) {
+            Some(upper_re) => upper_re,
+            None => return Ok(()), // The median can't be greater than u8::MAX.
```

Comment:
So we allow `2^256` milliseconds rounds?

Reply:
Yes. I don't think we gain anything by setting an explicit upper bound, because going too slow is not a viable attack.

Reply:
Like I asked in my other comment – doesn't it open up an attack where whale validators can force the very high round exponents on the rest?